File: app/src/debug/java/com/redditapp/ActivityLifecycleObserver.java
package com.redditapp;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import timber.log.Timber;

/** A "view server" adaptation which automatically hooks itself up to all activities. */
public class ActivityLifecycleObserver implements Application.ActivityLifecycleCallbacks {
    @Override public void onActivityCreated(Activity activity, Bundle bundle) {
        Timber.d("%s onActivityCreated", activity.getClass().getSimpleName());
    }
    @Override public void onActivityStarted(Activity activity) {
        Timber.d("%s onActivityStarted", activity.getClass().getSimpleName());
    }
    @Override public void onActivityResumed(Activity activity) {
        Timber.d("%s onActivityResumed", activity.getClass().getSimpleName());
    }
    @Override public void onActivityPaused(Activity activity) {
        Timber.d("%s onActivityPaused", activity.getClass().getSimpleName());
    }
    @Override public void onActivityStopped(Activity activity) {
        Timber.d("%s onActivityStopped", activity.getClass().getSimpleName());
    }
    @Override public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {
        Timber.d("%s onActivitySaveInstanceState", activity.getClass().getSimpleName());
    }
    @Override public void onActivityDestroyed(Activity activity) {
        Timber.d("%s onActivityDestroyed", activity.getClass().getSimpleName());
    }
}

File: app/src/debug/java/com/redditapp/ActivityLifecycleObserver.kt
package com.redditapp

import android.app.Activity
import android.app.Application
import android.os.Bundle

import timber.log.Timber

/** A "view server" adaptation which automatically hooks itself up to all activities.  */
class ActivityLifecycleObserver : Application.ActivityLifecycleCallbacks {
    override fun onActivityCreated(activity: Activity, bundle: Bundle) {
        Timber.d("%s onActivityCreated", activity.javaClass.simpleName)
    }

    override fun onActivityStarted(activity: Activity) {
        Timber.d("%s onActivityStarted", activity.javaClass.simpleName)
    }

    override fun onActivityResumed(activity: Activity) {
        Timber.d("%s onActivityResumed", activity.javaClass.simpleName)
    }

    override fun onActivityPaused(activity: Activity) {
        Timber.d("%s onActivityPaused", activity.javaClass.simpleName)
    }

    override fun onActivityStopped(activity: Activity) {
        Timber.d("%s onActivityStopped", activity.javaClass.simpleName)
    }

    override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) {
        Timber.d("%s onActivitySaveInstanceState", activity.javaClass.simpleName)
    }

    override fun onActivityDestroyed(activity: Activity) {
        Timber.d("%s onActivityDestroyed", activity.javaClass.simpleName)
    }
}